faf-java-server-app/src/test/java/com/faforever/server/game/SpoofDetectorServiceTest.java

```diff
@@ -0,0 +1,286 @@
+package com.faforever.server.game;
+
+import com.faforever.server.client.ClientService;
+import com.faforever.server.entity.Avatar;
+import com.faforever.server.entity.AvatarAssociation;
+import com.faforever.server.entity.Game;
+import com.faforever.server.entity.GlobalRating;
+import com.faforever.server.entity.Player;
+import com.faforever.server.error.ErrorCode;
+import com.faforever.server.player.PlayerService;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import static com.faforever.server.error.RequestExceptionWithCode.requestExceptionWithCode;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SpoofDetectorServiceTest {
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  private SpoofDetectorService instance;
+
+  @Mock
+  private PlayerService playerService;
+  @Mock
+  private ClientService clientService;
+
+  @Before
+  public void setUp() throws Exception {
+    instance = new SpoofDetectorService(playerService, clientService);
+  }
+
+  @Test
+  public void verifyPlayerReporterNotInGame() {
+    expectedException.expect(requestExceptionWithCode(ErrorCode.THIS_PLAYER_NOT_IN_GAME));
+
+    instance.verifyPlayer(new Player(), 1, "", 1, 1, null, null, null);
+    verifyZeroInteractions(clientService);
+  }
+
+  @Test
+  public void verifyPlayerPlayerNotOnline() {
+    when(playerService.getOnlinePlayer(1)).thenReturn(Optional.empty());
+    expectedException.expect(requestExceptionWithCode(ErrorCode.PLAYER_NOT_ONLINE));
+
+    instance.verifyPlayer(new Player().setCurrentGame(new Game()), 1, "", 1, 1, null, null, null);
+    verifyZeroInteractions(clientService);
+  }
+
+  @Test
+  public void verifyPlayerReporteeNotInGame() {
+    when(playerService.getOnlinePlayer(1)).thenReturn(Optional.of(new Player()));
+    expectedException.expect(requestExceptionWithCode(ErrorCode.OTHER_PLAYER_NOT_IN_GAME));
+
+    instance.verifyPlayer(new Player().setCurrentGame(new Game()), 1, "", 1, 1, null, null, null);
+    verifyZeroInteractions(clientService);
+  }
+
+  @Test
+  public void verifyPlayer() {
```

Comment:
whats the difference between this test and `verifyPlayerReporterAndReporteeNotInSameGame`

Reply:
nothing - good catch, I removed it :)